src/query/validation.c

```diff
@@ -240,19 +242,19 @@ static void verify_bucket_expression(Node *node)
 
     for (int i = 1; i < list_length(func_expr->args); i++)
     {
-      if (!IsA(unwrap_cast((Node *)list_nth(func_expr->args, i)), Const))
-        FAILWITH_LOCATION(func_expr->location, "Non-primary arguments for a generalization function have to be simple constants.");
+      Node *arg = unwrap_cast((Node *)list_nth(func_expr->args, i));
+      if (!is_stable_expression(arg))
+        FAILWITH_LOCATION(exprLocation(arg), "Non-primary arguments for a bucket function have to be simple constants.");
     }
   }
   else if (IsA(node, OpExpr))
   {
     OpExpr *op_expr = (OpExpr *)node;
     FAILWITH_LOCATION(op_expr->location, "Use of operators for generalization is not supported.");
   }
-  else if (IsA(node, Const))
+  else if (is_stable_expression(node))
   {
-    Const *const_expr = (Const *)node;
-    FAILWITH_LOCATION(const_expr->location, "Simple constants are not allowed as generalization expressions.");
+    FAILWITH_LOCATION(exprLocation(node), "Simple constants are not allowed as bucket expressions.");
```

Comment:
This merge reverses the error messages to the previous format. We have moved to using "generalization expressions" in all error messages a while ago.

Reply:
BTW, would you be more open to having one `master` branch + a feature toggle to switch between Elm and Fir? This merge was pretty hard.

Reply:
I still feel that the current approach is better.
I think that the difficulties you encountered were aggravated by the approach you took in adding these changes. If the Elm specific changes would have been added first, then there might have been less conflicts to resolve during merging.
In any case, I expect that further development on Elm will cease soon and it will go into maintenance mode.

Reply:
But the Elm specific changes were added first :). Anyhow, let's see how things evolve.

Reply:
I see that they were added first to the `Elm` branch, but not to `master`, where the conflicts occurred.
The "[Allow parametrized generalization expressions](https://github.com/diffix/pg_diffix/pull/408/commits/ceefe6a173336a22947b2c72615d5af068dffb71)" commit was pushed before the merge, not afterwards.